lib/internal/fs/utils.js

```diff
@@ -650,29 +651,103 @@ function warnOnNonPortableTemplate(template) {
   }
 }
 
+const defaultRmOptions = {
+  recursive: false,
+  force: false,
+  retryDelay: 100,
+  maxRetries: 0
+};
+
 const defaultRmdirOptions = {
   retryDelay: 100,
   maxRetries: 0,
   recursive: false,
 };
 
-const validateRmdirOptions = hideStackFrames((options) => {
-  if (options === undefined)
-    return defaultRmdirOptions;
-  if (options === null || typeof options !== 'object')
-    throw new ERR_INVALID_ARG_TYPE('options', 'object', options);
+const validateRmOptions = hideStackFrames((path, options, callback) => {
+  try {
+    options = validateRmdirOptions(options, defaultRmOptions);
+  } catch (err) {
+    return callback(err);
+  }
 
-  options = { ...defaultRmdirOptions, ...options };
+  if (typeof options.force !== 'boolean')
+    return callback(
+      new ERR_INVALID_ARG_TYPE('force', 'boolean', options.force)
+    );
 
-  if (typeof options.recursive !== 'boolean')
-    throw new ERR_INVALID_ARG_TYPE('recursive', 'boolean', options.recursive);
+  lazyLoadFs().stat(path, (err, stats) => {
+    if (err && err.code === 'ENOENT') {
+      if (options.force) {
+        return callback(null, options);
+      }
+      return callback(err, options);
+    }
+
+    if (err) {
+      return callback(err);
+    }
+
+    if (stats.isDirectory() && !options.recursive) {
+      return callback(new ERR_FS_EISDIR({
+        code: 'EISDIR',
+        message: 'is a directory',
+        path,
+        syscall: 'rm',
+        errno: -21
+      }));
+    }
+    return callback(null, options);
+  });
+});
 
-  validateInt32(options.retryDelay, 'retryDelay', 0);
-  validateUint32(options.maxRetries, 'maxRetries');
+const validateRmOptionsSync = hideStackFrames((path, options) => {
+  options = validateRmdirOptions(options, defaultRmOptions);
+
+  if (typeof options.force !== 'boolean')
+    throw new ERR_INVALID_ARG_TYPE('force', 'boolean', options.force);
+
+  try {
+    const stats = lazyLoadFs().statSync(path);
+
+    if (stats.isDirectory() && !options.recursive) {
+      throw new ERR_FS_EISDIR({
+        code: 'EISDIR',
+        message: 'is a directory',
+        path,
+        syscall: 'rm',
+        errno: -21
+      });
+    }
+  } catch (err) {
+    if (err.code !== 'ENOENT') {
+      throw err;
+    } else if (err.code === 'ENOENT' && !options.force) {
+      throw err;
+    }
```

Comment:
```suggestion
    if (!options.force || err.code !== 'ENOENT') {
      throw err;
    }
```

Reply:
This changes the functionality so that `force` suppresses _all_ errors vs. just `ENOENT`. The docs say we only suppress errors for nonexistent paths, which seems to be in line with `rm`.

Reply:
ah, fair, my bad